src/lug/Graphics/Vulkan/Gui.cpp

```diff
@@ -0,0 +1,876 @@
+            /* subpassDescription.inputAttachments */{},
+            /* subpassDescription.colorAttachments */{ { colorAttachmentIndex, VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL } },
+            /* subpassDescription.resolveAttachments */{},
+            /* subpassDescription.depthStencilAttachment */{},
+            /* subpassDescription.preserveAttachments */{},
+        };
+
+        renderPassBuilder.addSubpass(subpassDescription);
+
+        VkResult result{VK_SUCCESS};
+        API::RenderPass renderPass;
+        if (!renderPassBuilder.build(renderPass, &result)) {
+            LUG_LOG.error("Gui::initPipeline: Can't create render pass: {}", result);
+            return false;
+        }
+
+        graphicsPipelineBuilder.setRenderPass(std::move(renderPass), 0);
+    }
+
+    VkResult result{VK_SUCCESS};
+    if (!graphicsPipelineBuilder.build(_pipeline, &result)) {
+        LUG_LOG.error("Gui::initPipeline: Can't create pipeline: {}", result);
+        return false;
+    }
+
+    return true;
+}
+
+bool Gui::initFramebuffers(const std::vector<API::ImageView>& imageViews) {
+    const API::RenderPass* renderPass = _pipeline.getRenderPass();
+
+    for (size_t i = 0; i < imageViews.size(); ++i) {
+        // Create depth buffer image view
+        API::Builder::Framebuffer framebufferBuilder(_renderer.getDevice());
+
+        framebufferBuilder.setRenderPass(renderPass);
+        framebufferBuilder.addAttachment(&imageViews[i]);
+        framebufferBuilder.setWidth(imageViews[i].getImage()->getExtent().width);
+        framebufferBuilder.setHeight(imageViews[i].getImage()->getExtent().height);
+
+        VkResult result{VK_SUCCESS};
+        if (!framebufferBuilder.build(_framesData[i].framebuffer, &result)) {
+            LUG_LOG.error("Forward::initFramebuffers: Can't create framebuffer: {}", result);
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool Gui::beginFrame(const lug::System::Time& elapsedTime) {
+    ImGuiIO& io = ImGui::GetIO();
+
+    io.DeltaTime = elapsedTime.getSeconds();
+
+    io.DisplaySize = ImVec2(_window.getWidth(), _window.getHeight());
+
+    uint32_t x;
+    uint32_t y;
+    _window.getMousePos(x, y);
+    io.MousePos = ImVec2(static_cast<float>(x), static_cast<float>(y));
+
+    io.MouseDown[0] = _window.isMousePressed(lug::Window::Mouse::Button::Left);
+    io.MouseDown[1] = _window.isMousePressed(lug::Window::Mouse::Button::Right);
+    io.MouseDown[2] = _window.isMousePressed(lug::Window::Mouse::Button::Middle);
+
+    ImGui::NewFrame();
+
+    return true;
+}
+
+bool Gui::endFrame(const std::vector<VkSemaphore>& waitSemaphores, uint32_t currentImageIndex) {
+    ImGui::Render();
+    FrameData& frameData = _framesData[currentImageIndex];
+
+    if (!frameData.fence.wait()) {
+        return false;
+    }
+    frameData.fence.reset();
+    updateBuffers(currentImageIndex);
+
+    API::Device &device = _renderer.getDevice();
+
+    API::QueueFamily* graphicsQueueFamily;
```

Comment:
You won't need this anymore when you will rename `_presentQueueFamily` in `_graphicsQueueFamily`